### PR TITLE
[GeoMechanicsApplication] Fix 1d consolidation test (#12030)

### LINKED ISSUE #12030: [GeoMechanicsApplication] Fix the 1D consolidation benchmark test and understand its results
@muhammedfurkanyilmaz asked a question about a 1D consolidation test from our test suite (test case `test_benchmark1_4` in `applications/GeomechanicsApplication/tests/test_benchmark_set_1.py`). In particular, he is confused about the units of the applied permeability. See [this discussion](https://github.com/KratosMultiphysics/Kratos/discussions/12022) for his original question and the replies.

When we had a first look into the test case, we found that it is currently skipped during the test runs. After removing the statement that skips the test, when we attempted to run the test, the following error was raised:
```
ERROR: test_benchmark1_4 (__main__.KratosGeoMechanicsBenchmarkSet1.test_benchmark1_4)
test 1D consolidation on elastic soil.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Checkouts\KratosProjects\alpha\bin\Debug\applications\GeoMechanicsApplication\tests\test_benchmark_set_1.py", line 157, in test_benchmark1_4
    stage.Run()
  File "C:\Checkouts\KratosProjects\alpha\bin\Debug\KratosMultiphysics\analysis_stage.py", line 48, in Run
    self.Initialize()
  File "C:\Checkouts\KratosProjects\alpha\bin\Debug\KratosMultiphysics\analysis_stage.py", line 83, in Initialize
    self._GetSolver().PrepareModelPart()
  File "C:\Checkouts\KratosProjects\alpha\bin\Debug\KratosMultiphysics\GeoMechanicsApplication\geomechanics_U_Pw_solver.py", line 105, in PrepareModelPart
    super().PrepareModelPart()
  File "C:\Checkouts\KratosProjects\alpha\bin\Debug\KratosMultiphysics\GeoMechanicsApplication\geomechanics_solver.py", line 210, in PrepareModelPart
    self._ExecuteCheckAndPrepare()
  File "C:\Checkouts\KratosProjects\alpha\bin\Debug\KratosMultiphysics\GeoMechanicsApplication\geomechanics_solver.py", line 425, in _ExecuteCheckAndPrepare
    check_and_prepare_model_process_geo.CheckAndPrepareModelProcess(self.main_model_part, params).Execute()
  File "C:\Checkouts\KratosProjects\alpha\bin\Debug\KratosMultiphysics\GeoMechanicsApplication\check_and_prepare_model_process_geo.py", line 39, in Execute
    domain_parts.append(self.main_model_part.GetSubModelPart(self.problem_domain_sub_model_part_list[i].GetString()))
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: Error: There is no sub model part with name "Soil_two_phase-auto-1" in model part "PorousDomain"
The following sub model parts are available:
	"Soil_undrained-auto-1"
	"Line_Load-auto-1"
	"BOTTOM"
	"SIDE"
	"porous_computational_model_part"

in kratos/sources/model_part.cpp:2398: ModelPart::ErrorNonExistingSubModelPart
```

We need to clean up the input files first. Next, we may start to analyze the produced results and check whether they make any sense from a physical point of view.

### PR DESCRIPTION
**Pull request by WPK4FEM**
**📝 Description**
Enabled 1D consolidation test.
Simplified stage setup ( only 1 .mdpa file and 1 material file remain )
Stages end exactly at the proposed dimensionless times.

README.md file added together with picture of the test geometry and boundary conditions.

### CODE REVIEW

File: applications/GeoMechanicsApplication/tests/1D-Consolidation_all_stages/ProjectParameters_stage2.json [hunk 1]
```diff
@@ -133,23 +133,19 @@
...
     "loads_process_list": [{
         "python_module": "apply_vector_constraint_table_process",
         "kratos_module": "KratosMultiphysics.GeoMechanicsApplication",
         "process_name":  "ApplyVectorConstraintTableProcess",
         "Parameters":    {
-            "model_part_name": "PorousDomain.Line_Load-auto-1",
+            "model_part_name": "PorousDomain.Line_load",
             "variable_name":   "LINE_LOAD",
             "active":          [false,true,false],
             "value":           [0.0,-1,0.0],
```
Comment (indigocoral) [inline applications/GeoMechanicsApplication/tests/1D-Consolidation_all_stages/ProjectParameters_stage2.json]: Just to be consistent with the other ProjectParameters files :)
```suggestion
            "value":           [0.0,-1.0,0.0],
```

File: applications/GeoMechanicsApplication/tests/1D-Consolidation_all_stages/README.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
+# 1D Consolidation
+
+This tests verifies that in a column of triangular plane strain elements, consolidation follows the solution given by Terzaghi in 1923.
+
+## Setup
+
+This test consists of 78 6-noded elements (using the SmallStrainUPwDiffOrderElement2D6N class) of b x h = 0.1 x 1.0 meter. A
+schematic representation can be found in the figure below:
+
+![MeshStructure](MeshStructure.svg)
```
Comment (indigocoral) [inline applications/GeoMechanicsApplication/tests/1D-Consolidation_all_stages/README.md:10]: The figure is really good. Thanks for putting it with the documentation. Maybe just a suggestion:
Do you think it makes sense if we also show the amount of the load on top of the load? And maybe also the dimension of the soil column on the figure? I think this would help to get most of the information needed by just looking at the figure.

File: applications/GeoMechanicsApplication/tests/1D-Consolidation_all_stages/README.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+| 1728000.0        | 2.0                 |
+| 4320000.0        | 5.0                 |
+| 8640000.0        | 10.0                |
+
+## Assertions
+
+The test asserts that the vertical distribution of pressure matches the given Terzaghi solution at all nodes at the end of every stage.
+
+## Bibliography
+Verruijt, A., 2001. [Soil Mechanics](https://ocw.tudelft.nl/wp-content/uploads/SoilMechBook.pdf). Delft University of Technology.
```
Comment (indigocoral) [inline applications/GeoMechanicsApplication/tests/1D-Consolidation_all_stages/README.md]: Good that you have added this! Thank you.

File: applications/GeoMechanicsApplication/tests/1D-Consolidation_all_stages/README.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+where the consolidation coefficient:
+
+$$ c_v = \frac{\kappa}{\mu ( \frac{1}{K + 4G/3} + \frac{n}{K_w})}$$
+
+The analytical solution given by Terzaghi reads:
+
+$$ p = \frac{4 p_0}{\pi} \sum_{j=1}^{\infty} \frac{(-1)^{j-1}}{2j-1} cos((2j-1) \frac{\pi y}{2 h}) exp(-(2j-1)^2 \frac{\pi^2 c_v t}{4 h^2}) $$
+
+Analysis stages are chosen such that at the endtimes of the stages and the corresponding dimensionless time parameter $\frac{c_v t}{h^2}$ follows the pattern of the table below.
```
Comment (indigocoral) [inline applications/GeoMechanicsApplication/tests/1D-Consolidation_all_stages/README.md]:
```suggestion
Analysis stages are chosen such that at the end times of the stages, the corresponding dimensionless time parameter $\frac{c_v t}{h^2}$ follows the pattern of the table below.
```

File: applications/GeoMechanicsApplication/tests/test_benchmark_set_1.py [hunk 5]
```diff
@@ -118,9 +118,6 @@ def test_benchmark1_3(self):
         reaction_force = [(cauchy_stress_at_load[0][3]+cauchy_stress_at_load[1][3]+cauchy_stress_at_load[2][3])/3*step_size for cauchy_stress_at_load in cauchy_stresses_at_load]
         cauchy_stress_tensor[min_distance_idxs]
 
-
-    # todo
-    @KratosUnittest.skip("test should be checked")
```
Comment (indigocoral) [inline applications/GeoMechanicsApplication/tests/test_benchmark_set_1.py:123]: Do we want to modify the Python test in this PR or will it be done another time? The parameters in this test can be better explained. Also, do we want to keep the test in this benchmark set?

File: applications/GeoMechanicsApplication/tests/1D-Consolidation_all_stages/README.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+The material is described using:
+-   A linear elastic plane stress model (LinearElasticPlaneStrain2DLaw)
+-   A Young's modulus $E = 1.0E3 \mathrm{[Pa]}$ with Poisson's ratio $\nu = 0.0 \mathrm{[-]}$.
+-   The soil and water density are specified, but irrelevant due to the absence of gravity. The porosity is 0.3.
+These material properties of the dry soil give a compression modulus $K = E / (3(1-2\nu)) = 3.33E2 \mathrm{[Pa]}$ and a shear modulus $G = E / (2( 1 + \nu )) = 5.0E2 \mathrm{[Pa]}$.
+-   The dynamic viscosity of water is given as $\mu = 1.0E-6 \mathrm{[Pa \cdot s]}$ and the intrinsic permeability of the soil as $\kappa = 1.17982E-15 \mathrm{[m^2]}$
+-   The bulk modulus of water $K_w = 2.0E9 \mathrm{[Pa]}$
+
+The differential equation for the 1 D consolidation column under constant load as presented by Verruijt (2001) is :
```
Comment (indigocoral) [inline applications/GeoMechanicsApplication/tests/1D-Consolidation_all_stages/README.md]:
```suggestion
The differential equation for the 1D consolidation column under constant load as presented by Verruijt (2001) is :
```

File: applications/GeoMechanicsApplication/tests/1D-Consolidation_all_stages/README.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+## Setup
+
+This test consists of 78 6-noded elements (using the SmallStrainUPwDiffOrderElement2D6N class) of b x h = 0.1 x 1.0 meter. A
+schematic representation can be found in the figure below:
+
+![MeshStructure](MeshStructure.svg)
+
+All nodes on the sides have fixed horizontal displacements. The bottom nodes have fixed displacement in both the horizontal and the vertical
+direction. At the top of the column a vertical compressive line load with a magnitude of 1.0 N/m is applied. This load is kept constant during the whole analysis time. Note that gravity is absent in this computation, the top load generates the excess pore pressure. In the first stage no Dirichlet boundaries for the water pressure D.O.F. are given, such that all water is contained within the column. In all later stages, the water pressure on the top of the column is specified to be 0. [Pa]. This creates an outflow boundary at the top.
```
Comment (indigocoral) [inline applications/GeoMechanicsApplication/tests/1D-Consolidation_all_stages/README.md]: I think it would be better to also mention here that "p" in the figure is the water pressure.
```suggestion
direction. At the top of the column a vertical compressive line load with a magnitude of 1.0 N/m is applied. This load is kept constant during the whole analysis time. Note that gravity is absent in this computation, the top load generates the excess pore pressure. In the first stage no Dirichlet boundaries for the water pressure D.O.F. are given, such that all water is contained within the column. In all later stages, the water pressure on the top of the column (p) is specified to be 0 [Pa]. This creates an outflow boundary at the top.
```

File: applications/GeoMechanicsApplication/tests/1D-Consolidation_all_stages/ProjectParameters_stage2.json [hunk 8]
```diff
@@ -13,10 +13,10 @@
...
             "input_type":       "mdpa",
-            "input_filename":   "1D-Consolidationtest_stage2"
+            "input_filename":   "1D-Consolidationtest"
         },
         "material_import_settings":              {
-            "materials_filename":       "MaterialParameters_stage2.json"
+            "materials_filename":       "MaterialParameters.json"
         },
         "time_stepping":              {
             "time_step":            100
```
Comment (indigocoral) [inline applications/GeoMechanicsApplication/tests/1D-Consolidation_all_stages/ProjectParameters_stage2.json:22]: Might be a stupid question but is there a reason that the time step is different in every stage?

**Review by indigocoral — COMMENTED** [review]
Thank you @avdg81 and @WPK4FEM for fixing the test, adding documentation and also removing the unnecessary files. Great work. I only have minor suggestions and also, a question.

File: applications/GeoMechanicsApplication/tests/1D-Consolidation_all_stages/ProjectParameters_stage2.json (hunk 1 above)
Comment (WPK4FEM) [reply applications/GeoMechanicsApplication/tests/1D-Consolidation_all_stages/ProjectParameters_stage2.json]: Indeed, done.

File: applications/GeoMechanicsApplication/tests/1D-Consolidation_all_stages/README.md (hunk 4 above)
Comment (WPK4FEM) [reply applications/GeoMechanicsApplication/tests/1D-Consolidation_all_stages/README.md]: Changed to end times, but I tend to disagree with changing and to comma.

File: applications/GeoMechanicsApplication/tests/1D-Consolidation_all_stages/README.md (hunk 6 above)
Comment (WPK4FEM) [reply applications/GeoMechanicsApplication/tests/1D-Consolidation_all_stages/README.md]: done.

File: applications/GeoMechanicsApplication/tests/1D-Consolidation_all_stages/README.md (hunk 7 above)
Comment (WPK4FEM) [reply applications/GeoMechanicsApplication/tests/1D-Consolidation_all_stages/README.md]: Mentioned: the water pressure (p) on the top
changed 0. to 0.0 to match the 1.0 earlier.

File: applications/GeoMechanicsApplication/tests/1D-Consolidation_all_stages/ProjectParameters_stage2.json (hunk 8 above)
Comment (WPK4FEM) [reply applications/GeoMechanicsApplication/tests/1D-Consolidation_all_stages/ProjectParameters_stage2.json:22]: It is different because less and less is happening, therefore bigger steps are taken. See the analytical solution for the exponential decay. Not a stupid question, but a sign to thorough inspection.

File: applications/GeoMechanicsApplication/tests/1D-Consolidation_all_stages/README.md (hunk 2 above)
Comment (WPK4FEM) [reply applications/GeoMechanicsApplication/tests/1D-Consolidation_all_stages/README.md:10]: tried to change according to the comments. Please check again.

File: applications/GeoMechanicsApplication/tests/test_benchmark_set_1.py (hunk 5 above)
Comment (WPK4FEM) [reply applications/GeoMechanicsApplication/tests/test_benchmark_set_1.py:123]: We started documenting this test due to a user question. Now that we have it running and the comparison with the Terzaghi solution is working, we would like to have it running nightly. therefore the activation. 

Moving it out of the non-descriptive benchmark set 1 is a good suggestion. Please show what can be better explained, the README.md file was created for that.

Comment (WPK4FEM) [reply applications/GeoMechanicsApplication/tests/test_benchmark_set_1.py:123]: Decided to do the split of and possible annotation/improvement of the python part later.

**Review by avdg81 — APPROVED** [review]
Thanks for thoroughly cleaning up, documenting and reviving this test. It is much more readable now and it is no longer skipped during automatic test runs. Well done!
